api/acl_test.go

```diff
@@ -269,3 +269,35 @@ func TestACL_OneTimeToken(t *testing.T) {
 	assert.NotNil(t, out3)
 	assert.Equal(t, out3.AccessorID, out.AccessorID)
 }
+
+func TestACLTokens_BootstrapInvalidToken(t *testing.T) {
+	testutil.Parallel(t)
+	c, s := makeClient(t, nil, func(c *testutil.TestServerConfig) {
+		c.ACL.Enabled = true
+	})
+	defer s.Stop()
+	at := c.ACLTokens()
+
+	bootkn := &BootstrapRequest{}
+	bootkn.Secret = "badtoken"
+	// Bootstrap with invalid token
+	_, _, err := at.Bootstrap(bootkn, nil)
+	assert.EqualError(t, err, "Unexpected response code: 400 (invalid acl token: <nil>)")
+}
+
+func TestACLTokens_BootstrapValidToken(t *testing.T) {
+	testutil.Parallel(t)
+	c, s := makeClient(t, nil, func(c *testutil.TestServerConfig) {
+		c.ACL.Enabled = true
+	})
+	defer s.Stop()
+	at := c.ACLTokens()
+
+	bootkn := &BootstrapRequest{}
+	bootkn.Secret = "2b778dd9-f5f1-6f29-b4b4-9a5fa948757a"
+	// Bootstrap with Valid token
+	out, wm, err := at.Bootstrap(bootkn, nil)
+	assert.Nil(t, err)
```

Comment:
This gives a bit nicer error message on test failure:

```suggestion
	assert.NoError(t, err)
```